In [1]:
from utils import *
from Feature_extraction import *
from classical_models import *

In [2]:
# train_test_split()

In [3]:
train = pd.read_csv('train.csv')
test  = pd.read_csv('test.csv')
encode_category(train,test)

In [4]:
train['story'] =train['story'] .apply(preprocess_arabic_text)
test['story'] =test['story'] .apply(preprocess_arabic_text)

In [5]:
x_train,x_test=TFIDF(train['story'],test['story'])
y_train=train['topic']
y_test=test['topic']

In [6]:
# GridSearch(x_train,y_train,x_test,y_test)

In [11]:
y_pred=svm_model(x_train,y_train,x_test)
get_metrics(y_test,y_pred)

Overall Metrics:
Precision (weighted): 0.836
Recall (weighted): 0.832
F1 Score (weighted): 0.833
Accuracy: 0.832

Metrics for Each Class:
              precision    recall  f1-score   support

           0       0.81      0.94      0.87       200
           1       0.82      0.85      0.84       200
           2       0.94      0.93      0.93       200
           3       0.83      0.88      0.85       200
           4       0.96      0.84      0.90       200
           5       0.57      0.57      0.57       200
           6       0.73      0.80      0.76       200
           7       0.85      0.84      0.85       200
           8       0.72      0.65      0.68       200
           9       0.99      0.93      0.96       200
          10       0.97      0.92      0.95       200

    accuracy                           0.83      2200
   macro avg       0.84      0.83      0.83      2200
weighted avg       0.84      0.83      0.83      2200



(0.8362110634463815,
 0.8322727272727273,
 0.8328458816783101,
 0.8322727272727273)

>|Model |Paramerters |Precision| Recall|F1 SCORE |  Accuracy|
>|----|----|----|----|----|----|
>|SVM|    kernel=rbf  |  0.836|   0.832| 0.833 |0.832|
>|SVM|    kernel=linear  |  0.826|   0.826|  0.825 | 0.826|
>|Linear Regression|    - |0.830 | 0.829|   0.827|  0.829|
>|XGboost|   - |  0.799|  0.801|0.798 |0.801|
>|Descision Tree|    - |0.638  |  0.639|  0.637|  0.639|






> #### _`precision:`_  $  \frac  {tp}{tp+fp} $
> it is more concerned with how many of the classified topics are the trie ones without falsely classifying some into the wrong class 
>
> #### _`recall:`_  $  \frac  {tp}{tp+fn}$ 
> it is more concerned with how many of the true topics have i correclty noted as related to that class of topic
>
> #### _`f-score:`_ $  \frac  {precision *  recall}{precision +  recall}$
> it is a good mertic to use in case of imbalance in class , which is not our case here because the classes are balanced
>
> #### _`accuracy:`_ $  \frac  {tp+tn}{tp+fp+fn+tn}$
> since we have balanced class , it refers how many correct topics have i classified, so i correctly noticed that it is lying in topic1 and noticed that it doesnot lie in topic2


 
